### PR TITLE
1430 new project sort order

### LINKED ISSUE #1430: Sort project page by commit date
As a user
In order to see which projects have been worked on lately
I would like to order the projects by last worked on date

### PR DESCRIPTION
**Pull request by mattlindsey**
#### DEPLOYMENT NOTE
The nightly github status heroku job should be run immediately after deploying this so that the sort order of projects is correct.

#### Description
Re-orders the 'Projects' page by last_update in Github so that most recently active projects come to the top.

#### Motivation and Context
Makes website more user friendly.
fixes #1430 

#### How Has This Been Tested?
Tested by updating the project field 'last_github_update' by running the nightly update job
and seeing that the projects are sorted in correct order.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class ProjectsController < ApplicationController
 #TODO YA Add controller specs for all the code
 
   def index
-    @projects = Project.order('status ASC').order('commit_count DESC NULLS LAST').search(params[:search], params[:page]).includes(:user)
+    @projects = Project.order('status ASC').order('last_github_update DESC NULLS LAST').order('commit_count DESC NULLS LAST').search(params[:search], params[:page]).includes(:user)
```
Comment (mattlindsey) [inline app/controllers/projects_controller.rb]: I realize that this line kinda 'rigs' the test to always work, but I don't think there's truely a way to test using the existing mocking.

File: spec/controllers/projects_controller_spec.rb [hunk 2]
```diff
@@ -23,30 +23,28 @@
...
       expect(assigns(:projects).title).to eq 'Carrier has arrived.'
     end
 
-    #TODO: Refactor! This test is wrong. I can not test the order of projects.
-    it 'orders project by commit_count' do
-      allow_message_expectations_on_nil
+    it 'orders project by github update' do
       allow(Project).to receive(:search).and_return(@projects)
-      @projects.stub(:includes).and_return([@project2, @project])
+      @projects.stub(:includes).and_return([@project, @project2].sort {|a,b| b.last_github_update <=> a.last_github_update})
```
Comment (mattlindsey) [inline spec/controllers/projects_controller_spec.rb]: I realize that this line kinda 'rigs' the test to always work, but I don't think there's truely a way to test using the existing mocking.

File: app/controllers/projects_controller.rb (hunk 1 above)
Comment (tansaku) [inline app/controllers/projects_controller.rb]: could we use scopes here to reduce method length?

```rb
Project.by_status.by_github_update.by_commit_count.search ...
```

???

File: spec/controllers/projects_controller_spec.rb (hunk 2 above)
Comment (tansaku) [inline spec/controllers/projects_controller_spec.rb]: I think maybe we should delete the spec and do a cucumber test

File: app/controllers/projects_controller.rb (hunk 1 above)
Comment (mattlindsey) [inline app/controllers/projects_controller.rb]: I think this is just as clear as scopes for now, but let me know if you really want to change.

File: features/projects/list_projects.feature [hunk 3]
```diff
@@ -60,29 +59,24 @@ Feature: Browse and create projects
...
-#    And I should not see "greetings earthlings"
-#    When I go to the next page
-#    Then I should not see:
-#      | greetings aliens        |
-#      | greetings jupiter folks |
-#      | greetings mercury folks |
-#      | greetings saturn folks  |
-#      | greetings sun folks     |
-#    And I should see "greetings earthlings"
+  Scenario: Ordered display pagination in "Our Projects" page
```
Comment (tansaku) [inline features/projects/list_projects.feature]: great job getting this working, although would be good if we could express the intent, e.g. that we want to see the most recently updated projects at the top ...

Comment (tansaku) [inline features/projects/list_projects.feature]: feel free to push that onto me if you're losing motivation to keep tweaking this :-) could go in a refactoring ticket ...

**Pull request merged**